### PR TITLE
Setup test environment

### PR DESCRIPTION
**Pull request by onyxvd**
Create first Happy Path test to validate basic monitoring creation flow

### CODE REVIEW

File: app/src/androidTest/java/org/bspb/smartbirds/pro/LoginActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Rule
+    @JvmField
+    val batteryNotificationRule = SmartbirdsStateRule.setBatteryNotification(true)
+
+    @Rule
+    @JvmField
+    var screenshotRule = RuleChain
+            .outerRule(GrantPermissionRule.grant(permission.WRITE_EXTERNAL_STORAGE))
+            .around(ScreenshotTestRule())
```
Comment (groupsky) [inline app/src/androidTest/java/org/bspb/smartbirds/pro/LoginActivityTest.kt:48]: As this is replicated across tests, it seems better to provide rule with the chain inside

File: app/src/androidTest/java/org/bspb/smartbirds/pro/LoginActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        loginScreen {
+            isDisplayed()
+            fieldUsername().perform(typeText("user@smartbirds.org"))
+            fieldPassword().perform(typeText("Secret1!"), closeSoftKeyboard())
+            buttonLogin().perform(scrollTo(), click())
+        }
+
+        mainScreen {
+            isDisplayed()
```
Comment (groupsky) [inline app/src/androidTest/java/org/bspb/smartbirds/pro/LoginActivityTest.kt:72]: How would one check the opposite - that the screen is not displayed?

File: app/src/androidTest/java/org/bspb/smartbirds/pro/MonitoringTest.kt [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        }
+        commonFormScreen {
+            buttonSubmit().perform(click())
+        }
+
+
+        // Open birds form
+        monitoringScreen {
+            isDisplayed()
+            Thread.sleep(250)
```
Comment (groupsky) [inline app/src/androidTest/java/org/bspb/smartbirds/pro/MonitoringTest.kt:104]: Why the sleep?

File: app/src/androidTest/java/org/bspb/smartbirds/pro/MonitoringTest.kt [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            locationRule.updateLocation()
+
+            // Wait
+            Thread.sleep(1000)
```
Comment (groupsky) [inline app/src/androidTest/java/org/bspb/smartbirds/pro/MonitoringTest.kt:108]: same - why sleep?

File: app/src/androidTest/java/org/bspb/smartbirds/pro/tools/robot/AlertDialogTestRobot.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        fun alertDialog(block: AlertDialogTestRobot.() -> Unit): AlertDialogTestRobot {
+            return AlertDialogTestRobot().apply(block)
+        }
+    }
+
+    fun isDisplayed(@StringRes resource: Int): ViewInteraction =
+        onView(withText(resource)).check(matches(ViewMatchers.isDisplayed()))
+
+    fun button1(): ViewInteraction = onView(withId(android.R.id.button1))
+    fun button(): ViewInteraction = onView(withId(android.R.id.button2))
```
Comment (groupsky) [inline app/src/androidTest/java/org/bspb/smartbirds/pro/tools/robot/AlertDialogTestRobot.kt:27]:
```suggestion
    fun button2(): ViewInteraction = onView(withId(android.R.id.button2))
```

File: app/src/androidTest/java/org/bspb/smartbirds/pro/tools/rule/MockBackendRule.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                server.start()
+
+                // replace retrofit proxy with our mock
+                Backend.backendBaseUrl =
+                    "http://" + server.hostName + ":" + server.port
+//                                api = Mockito.mock(SmartBirdsApi.class);
+//                when(api.login(any())).then(invocation -> {
+//                    return new Call<LoginResponse>() {}
+//                })
+//                        .spy(Backend_.getInstance_(getApplicationContext()).api());
```
Comment (groupsky) [inline app/src/androidTest/java/org/bspb/smartbirds/pro/tools/rule/MockBackendRule.kt:32]: commented code

File: app/src/main/kotlin/org/bspb/smartbirds/pro/sync/UploadManager.kt [hunk 7]
```diff
@@ -53,11 +54,9 @@ open class UploadManager {
     @Bean
     protected lateinit var monitoringManager: MonitoringManager
 
-    @StringRes(R.string.tag_lat)
-    protected lateinit var tagLatitude: String
+    protected var tagLatitude: String = "latitude"
 
-    @StringRes(R.string.tag_lon)
-    protected lateinit var tagLongitude: String
+    protected var tagLongitude: String = "longitude"
```
Comment (groupsky) [inline app/src/main/kotlin/org/bspb/smartbirds/pro/sync/UploadManager.kt:59]: Seems the change is out of scope of the PR

File: app/build.gradle [hunk 8]
```diff
@@ -89,6 +96,16 @@ android {
     kapt {
         correctErrorTypes = true
     }
+
+    testOptions {
+        animationsDisabled = true
```
[the remark this answers is not in the document]
Comment (onyxvd) [reply app/build.gradle:101]: no, this is used for a different thing

File: .github/workflows/tests.yml [hunk 9]
```diff
@@ -0,0 +1,28 @@
+name: Android CI Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
[the remark this answers is not in the document]
Comment (onyxvd) [reply .github/workflows/tests.yml:7]: https://trello.com/c/QaDnOasW/173-check-github-actions-limits-and-enable-project-actions-for-all-branches-if-no-limit-issues

File: app/src/main/kotlin/org/bspb/smartbirds/pro/sync/UploadManager.kt (hunk 7 above)
Comment (onyxvd) [reply app/src/main/kotlin/org/bspb/smartbirds/pro/sync/UploadManager.kt:59]: https://trello.com/c/N6q7mARn/174-fix-coordinates-null-check-in-uploadmanager

File: app/src/androidTest/java/org/bspb/smartbirds/pro/LoginActivityTest.kt (hunk 2 above)
Comment (onyxvd) [reply app/src/androidTest/java/org/bspb/smartbirds/pro/LoginActivityTest.kt:68]: This is out of scope. Will figure it out later.
